### PR TITLE
Feat: client side routing

### PR DESCRIPTION
**Pull request by tinsaye**
Adds some basic client side routing. 

pls, don't blame my CSS.

Suggested dashboard changes, under `Paths`:
clerk

### CODE REVIEW

File: src/client/src/auth/RautGuard.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { Outlet, useNavigate } from "@solidjs/router";
+import { createEffect } from "solid-js";
+import Header from "../components/Header";
+import clerk from "./Auth";
+
+const RouteGuard = () => {
```
Comment (cgsdev0) [inline src/client/src/auth/RautGuard.tsx]: q: why is the component called `RouteGuard` but the file is called `RautGuard`?

Comment (tinsaye) [reply src/client/src/auth/RautGuard.tsx]: ups, typing is hard

File: src/client/src/404.tsx [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export default function NotFound() {
+  return (
+    <div>
+      <h1>404</h1>
+      <p>Book not found.</p>
+    </div>
+  );
+}
```
Comment (twny) [inline src/client/src/404.tsx:8]: not found component is great!

File: src/client/src/App.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      <Route path="/" component={Home}/>
+      <Route path="/home" component={Home}/>
+      <Route path="/sign-up" component={SignUp}/>
+      <Route path="/sign-in" component={SignIn}/>
+      <Route path="/" component={RouteGuard}>
+        <Route path={"/dashboard"} component={Dashboard}/>
+        <Route path={"/payment"} component={Stripe}/>
+      </Route>
+      <Route path="*" component={NotFound}/>
+    </Routes>
```
Comment (twny) [inline src/client/src/App.tsx:23]: great work setting up routes!!

File: src/client/src/auth/RouteGuard.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+  return (
+    <div>
+      <Header/>
+      <Outlet/>
+    </div>
+  )
+}
+
+export default RouteGuard;
```
Comment (twny) [inline src/client/src/auth/RouteGuard.tsx:23]: very useful!

**Review by twny — APPROVED** [review]
i love this PR!
lets shipit

File: src/client/src/auth/RautGuard.tsx (hunk 1 above)
Comment (twny) [reply src/client/src/auth/RautGuard.tsx]: nice catch!

**Pull request merged**